Subscribers/CronJob.php

```diff
@@ -228,9 +243,26 @@ private function removeArticlesFromStream(array $articleIds)
     private function getStreamService()
     {
         if (!$this->streamService) {
-            $this->streamService = Shopware()->Container()->get('swagconnect.product_stream_service');
+            $this->streamService = $this->container->get('swagconnect.product_stream_service');
         }
 
         return $this->streamService;
     }
+
+    /**
+     * @return ProductSearch
+     */
+    private function getProductSearch()
+    {
+        if (!$this->productSearch) {
+            // HACK
+            // do not use as a dependency!!!
+            // this class uses Shopware product search which depends on shop context
+            // so if it's used as dependency of subscriber, plugin returns error on deactivate
+            // see CON-4922
+            $this->productSearch = $this->container->get('swagconnect.product_search');
```

Comment:
if this works fine -> but shoudn't the streamservice and thumbnail manager be injected in the constructor?
Imho it's really bad to use the container because your dependencies are implicit so i would suggest to put as much in the constructor as possible